File: src/common/web/data/entities/project/logbook/PublishingJobRecord.ts
import { type ConnectorInstanceID } from "../../connector/ConnectorInstance";
import { ProjectLogbookRecord } from "./ProjectLogbookRecord";

/**
 * A single record of a project's running publication job.
 *
 * @param connector_instance - The connector instance ID.
 * @param progress - The total progress (0.0 - 1.0).
 * @param message - The current activity message.
 */
export class PublishingJobRecord extends ProjectLogbookRecord {
    public readonly connector_instance: ConnectorInstanceID;

    public readonly progress: number;
    public readonly message: string;

    public constructor(timestamp: number, connectorInstance: ConnectorInstanceID, progress: number, message: string) {
        super(timestamp);

        this.connector_instance = connectorInstance;

        this.progress = progress;
        this.message = message;
    }
}

File: src/common/py/data/entities/project/logbook/publishing_job_record.py
from dataclasses import dataclass

from dataclasses_json import dataclass_json

from .project_logbook_record import ProjectLogbookRecord
from ...connector import ConnectorInstanceID


@dataclass_json
@dataclass
class PublishingJobRecord(ProjectLogbookRecord):
    """
    A single record of a project's running publication job.

    Attributes:
        connector_instance: The connector instance ID.
        progress: The total progress (0.0 - 1.0).
        message: The current activity message.
    """

    connector_instance: ConnectorInstanceID

    progress: float
    message: str
